src/lib.rs

```diff
@@ -31,25 +36,41 @@ mod inner {
         const BLOCK_SIZE: u8 = 8;
 
         #[new]
-        #[pyo3(signature = (string = Vec::new()))]
-        fn new(string: Vec<u8>) -> Self {
-            SeaHash { inner: string }
+        #[pyo3(signature = (data = Vec::new()))]
+        fn new(data: Vec<u8>) -> Self {
+            let mut inner = seahash::SeaHasher::new();
+            inner.write(data.as_slice());
+            SeaHash { inner }
         }
 
+        #[cfg(not(Py_3_11))]
         pub fn update(&mut self, obj: &[u8]) {
-            self.inner.extend_from_slice(obj)
+            self.inner.write(obj)
+        }
+
+        #[cfg(Py_3_11)]
+        pub fn update(&mut self, py: Python, obj: &PyAny) -> PyResult<()> {
+            match obj.extract() {
+                Ok(buf) => {
+                    self.inner.write(buf);
+                }
+                Err(_) => {
+                    self.inner.write(PyBuffer::get(obj)?.to_vec(py)?.as_slice());
```

Comment:
Question: Could we use overloading instead to get the correct type for this? There's 2 fairly separate implementations here.

Reply:
Rust doesn't support function overloading... Maybe PyO3 can help out here? I'm not sure how to achieve this.